### PR TITLE
[NFC][SYCL] Update tests to reflect new offload model

### PR DESCRIPTION
**Pull request by mdtoguchi**
Update existing SYCL based driver tests to use the new offload model and retain the equivalent old offloading model tests be creating copies. These tests use the --offload-new-driver and --no-offload-new-driver options to force a particular model to follow.

### CODE REVIEW

File: clang/test/Driver/clang-offload-bundler-bc-archive-support-linux.cpp [hunk 1]
```diff
@@ -10,104 +11,104 @@
...
-////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make bundled object with targets:
+// sycl-spir64-unknown-unknown
+// host-x86_64-unknown-linux-gnu
+// RUN: %clangxx -fsycl --no-offload-new-driver -c %s -o %t_bundled.o
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Make three distinct BC files
+// RUN: %clangxx -fsycl -fsycl-device-only -DTYPE1 %s -o %t1.bc
```
Comment (srividya-sundaram) [inline clang/test/Driver/clang-offload-bundler-bc-archive-support-linux.cpp:22]: Is `--no-offload-new-driver` not required here?
I see it added [here](https://github.com/intel/llvm/pull/14539/files?diff=unified&w=0#diff-2691373ddc872311fdaadae7dbfe0eeeaaab35a1fa7b3aa3a145e6d49017535aR22) for generating BC files.

File: clang/test/Driver/ftarget-compile-fast-old-model.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Test -ftarget-compile-fast behaviors
```
Comment (srividya-sundaram) [inline clang/test/Driver/ftarget-compile-fast-old-model.cpp:1]: Some of the comments are updated with // Old offloading model only
Can be made consistent by updating all comments as necessary.

File: clang/test/Driver/ftarget-compile-fast.cpp [hunk 3]
```diff
@@ -1,20 +1,21 @@
 // Test -ftarget-compile-fast behaviors
```
Comment (srividya-sundaram) [inline clang/test/Driver/ftarget-compile-fast.cpp:1]: Comment can be updated to mention the use of new offload driver

File: clang/test/Driver/sycl-add-default-spec-consts-image.cpp [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-// RUN: %clang -### -fsycl -fsycl-add-default-spec-consts-image -fsycl-targets=spir64 2>&1 %s | FileCheck %s -check-prefix=CHECK-NON-AOT
+// RUN: %clang -### -fsycl --no-offload-new-driver -fsycl-add-default-spec-consts-image 2>&1 %s | FileCheck %s -check-prefix=CHECK-NON-AOT
+// RUN: %clang_cl -### -fsycl --no-offload-new-driver -fsycl-add-default-spec-consts-image 2>&1 %s | FileCheck %s -check-prefix=CHECK-NON-AOT
+// RUN: %clang -### -fsycl --no-offload-new-driver -fsycl-add-default-spec-consts-image -fsycl-targets=spir64 2>&1 %s | FileCheck %s -check-prefix=CHECK-NON-AOT
 // CHECK-NON-AOT: warning: -fsycl-add-default-spec-consts-image flag has an effect only in Ahead of Time Compilation mode (AOT)
 
 // Check that non-AOT target doesn't add command line option into sycl-post-link invocation
 // CHECK-NON-AOT-NOT: {{.*}}sycl-post-link{{.*}} "-generate-device-image-default-spec-consts"
 
 // Check that no warnings are issued in correct cases and "-generate-device-image-default-spec-consts" is passed to sycl-post-link
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-add-default-spec-consts-image.cpp:13]: Comment update.

File: clang/test/Driver/sycl-device-lib-old-model.cpp [hunk 5]
```diff
@@ -0,0 +1,233 @@
+///
+/// Perform several driver tests for SYCL device libraries on Linux
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-lib-old-model.cpp:2]: Is there a test file for tests using the new offload driver?

File: clang/test/Driver/sycl-device-lib.cpp [hunk 6]
```diff
@@ -7,227 +7,184 @@
 /// ###########################################################################
 
 /// test behavior of device library default link and fno-sycl-device-lib-jit-link
-// RUN: %clangxx -fsycl %s --sysroot=%S/Inputs/SYCL -### 2>&1 \
+// RUN: %clangxx -fsycl --offload-new-driver %s --sysroot=%S/Inputs/SYCL -### 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-lib.cpp:10]: nit - filename can be updated to mention new offload model.

File: clang/test/Driver/sycl-device-old-model.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
+/// Check that compiling for sycl device is disabled by default:
+// RUN:   %clang -### %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK-DEFAULT %s
+// CHECK-DEFAULT-NOT: "-fsycl-is-device"
+
+/// Check "-fsycl-is-device" is passed when compiling for device:
+// RUN:   %clang -### -fsycl-device-only %s 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-old-model.cpp:7]: Should `--no-offload-new-driver` be passed here?

File: clang/test/Driver/sycl-device-old-model.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+// RUN:   | FileCheck -check-prefix=CHECK-SYCL-DEV %s
+// CHECK-SYCL-DEV: "-fsycl-is-device"{{.*}} "-internal-isystem" "{{.*}}bin{{[/\\]+}}..{{[/\\]+}}include{{[/\\]+}}sycl" "-internal-isystem" "{{.*}}bin{{[/\\]+}}..{{[/\\]+}}include"
+
+/// Check that "-Wno-sycl-strict" is set on compiler invocation with "-fsycl"
+/// or "-fsycl-device-only" or both:
+// RUN:   %clang -### -fsycl %s 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-old-model.cpp:13]: Same, should `--no-offload-new-driver` be used here?

File: clang/test/Driver/sycl-device-old-model.cpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+// CHECK-SYCL-STD_VERSION: clang{{.*}} "-sycl-std=2020"
+
+/// Check that -aux-triple is set correctly
+// RUN:   %clang -### -fsycl -target aarch64-linux-gnu %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK-SYCL-AUX-TRIPLE %s
+// TODO: %clang -### -fsycl -fsycl-device-only -target aarch64-linux-gnu
+// CHECK-SYCL-AUX-TRIPLE: clang{{.*}} "-aux-triple" "aarch64-unknown-linux-gnu"
+
+/// Verify output files are properly specified given -o
+// RUN: %clang -### -fsycl -fsycl-device-only -o dummy.out %s 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-old-model.cpp:33]: Same question as above.

File: clang/test/Driver/sycl-device-old-model.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+// RUN:   | FileCheck -check-prefix=CHECK-SYCL-NO_STRICT %s
+// CHECK-SYCL-NO_STRICT: clang{{.*}} "-Wno-sycl-strict"
+
+/// Check that -sycl-std=2017 is set if no std version is provided by user
+// RUN:   %clang -### -fsycl %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK-SYCL-STD_VERSION %s
+// CHECK-SYCL-STD_VERSION: clang{{.*}} "-sycl-std=2020"
+
+/// Check that -aux-triple is set correctly
+// RUN:   %clang -### -fsycl -target aarch64-linux-gnu %s 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-old-model.cpp:27]: Same question.

File: clang/test/Driver/sycl-device-optimizations.cpp [hunk 11]
```diff
@@ -1,11 +1,11 @@
 /// Check that optimizations for sycl device are enabled by default:
-// RUN:   %clang -### -fsycl %s 2>&1 \
+// RUN:   %clang -### -fsycl --offload-new-driver %s 2>&1 \
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-device-optimizations.cpp:2]: nit: All test files with the the old driver option have "old model" in their name.
Test files with the new driver model can have their names updated too.

Comment (mdtoguchi) [reply clang/test/Driver/sycl-device-optimizations.cpp:2]: I'd rather not shuffle the names around too much.  The idea is that the 'old model' will be going away once we make it the default, so the files renamed as 'old-model' can just be removed (as well as the existing tests that weren't converted to the new offloading model)

File: clang/test/Driver/sycl-device-lib-old-model.cpp (hunk 5 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-device-lib-old-model.cpp:2]: For this, no.  Any device libraries or objects aren't touched by the driver with the new model, so there aren't any behaviors to verify.

File: clang/test/Driver/ftarget-compile-fast-old-model.cpp (hunk 2 above)
Comment (mdtoguchi) [reply clang/test/Driver/ftarget-compile-fast-old-model.cpp:1]: The files that are commented with 'Old offload model only' are tests that already exist that were not converted for the new model as those behaviors being tested do not make sense for the new model.

File: clang/test/Driver/clang-offload-bundler-bc-archive-support-linux.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/test/Driver/clang-offload-bundler-bc-archive-support-linux.cpp:22]: It is required - will add.

File: clang/test/Driver/ftarget-compile-fast.cpp (hunk 3 above)
Comment (mdtoguchi) [reply clang/test/Driver/ftarget-compile-fast.cpp:1]: The files where an 'old-model' variant was added I do not update the comments to reflect.  The naming should take care of that.  Once the old model goes away, all of these tests will also be updated to remove the `--offload-new-driver` options being set.

File: clang/test/Driver/sycl-add-default-spec-consts-image.cpp (hunk 4 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-add-default-spec-consts-image.cpp:13]: Can you clarify?

File: clang/test/Driver/clang-offload-bundler-bc-archive-support-win.cpp [hunk 12]
```diff
@@ -10,105 +11,105 @@
...
 // RUN: clang-offload-bundler -unbundle -type=aoo -input=%t_bundled.a -targets=sycl-spir64-unknown-unknown,host-spir64_gen -output=%t_listA.txt -output=%t_listB.txt
-// Backtick not supported on Windows
-// RUN_disabled_because_backtick_not_supported: cmp %t_unbundled_A.o `cat %t_listA.txt`
-// RUN_disabled_because_backtick_not_supported: cmp %t2.bc           `cat %t_listB.txt`
+// Backtick not supported on Windows
+// RUN_disabled_because_backtick_not_supported: cmp %t_unbundled_A.o `cat %t_listA.txt`
+// RUN_disabled_because_backtick_not_supported: cmp %t2.bc           `cat %t_listB.txt`
 
-#include <sycl/sycl.hpp>
+#include <sycl/sycl.hpp>
```
Comment (mdtoguchi) [inline clang/test/Driver/clang-offload-bundler-bc-archive-support-win.cpp:112]: All of the non-visual changes are due to a cleanup of some carriage return characters.

File: clang/test/Driver/clang-offload-bundler-bc-archive-support2-linux.cpp [hunk 13]
```diff
@@ -1,40 +1,40 @@
...
+
+// RUN: wc %t_list.txt -l | FileCheck --check-prefixes=CHECK-LIST-LENGTH %s
+
+// CHECK-LIST-LENGTH: 2
+
+// RUN: cmp %t_unbundled.bc `grep .bc$ %t_list.txt`
+// RUN: cmp %t_unbundled.o  `grep .o$  %t_list.txt`
+
+void foo() {
+}
```
Comment (mdtoguchi) [inline clang/test/Driver/clang-offload-bundler-bc-archive-support2-linux.cpp:40]: Similar comment - non-visual changes to clean up carriage return characters.

File: clang/test/Driver/sycl-device-optimizations.cpp (hunk 11 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-device-optimizations.cpp:2]: I've thought about this a little, and I'm going to rename all of the tests that I'm not converting/adding to the new model to use 'old-model'.  This way, we know which ones will go away cleanly (have old-model in the name) and we don't have to worry about the comments.

**Review by srividya-sundaram — APPROVED** [review]

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, please take a look, should be ready for merge.

**Pull request merged**